lib/ansible/modules/clustering/kubevirt/k8s_service.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# Apache License, Version 2.0
+# (see LICENSE or http://www.apache.org/licenses/LICENSE-2.0)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: k8s_service
+
+short_description: Manage Services on Kubernetes
+
+description:
+    - Use Openshift Python SDK to manage Services on Kubernetes
+
+version_added: "2.8"
+
+author: KubeVirt Team (@kubevirt)
+
+options:
+    name:
+        description:
+            - Use to specify a Service object name.
+        required: true
+        type: str
+    namespace:
+        description:
+            - Use to specify a Service object namespace.
+        required: true
+        type: str
+    type:
+        description:
+            - Specifies the type of Service to create.
+        choices:
+            - NodePort
+            - ClusterIP
+            - LoadBalancer
+        required: true
+    ports:
+        description:
+            - A list of ports to expose.
+        required: true
+        type: list
+    selector:
+        description:
+            - Label selectors identify objects this Service should apply to.
+        default: {}
```

Comment:
I wonder whether it makes sense to create a service without selector `{}`. I think we should require it every time. Documentation suggest not to use selectors in:
> * You want to have an external database cluster in production, but in test you use your own databases.
> * You want to point your service to a service in another Namespace or on another cluster.
> * You are migrating your workload to Kubernetes and some of your backends run outside of Kubernetes.

Reply:
Yeah, I was wondering how to handle this. I think you're right, we should go with requiring a selector and if someone really really knows what they're doing, they can set an empty selector via definition.metadata.selector (that should work).